swarm/network/kademlia.go

```diff
@@ -335,66 +338,24 @@ func (k *Kademlia) On(p *Peer) (uint8, bool) {
 // sets it to the nDepth and sends a signal to every nDepthSig channel.
 func (k *Kademlia) setNeighbourhoodDepth() {
 	nDepth := depthForPot(k.conns, k.NeighbourhoodSize, k.base)
-	var changed bool
-	k.nDepthMu.Lock()
 	if nDepth != k.nDepth {
+		k.DepthChangeCond.L.Lock()
```

Comment:
Could you maybe use `k.lock` and not `k.DepthChangeCond.L` - you are protecting the `k.nDepth` change, and not the broadcast call, so I find it a bit confusing.